src/types/resolveDescriptors.ts

```diff
@@ -2237,22 +2236,24 @@ function initializeConstantsAndDefaultContractAndStructFields(
                             field.default = evalConstantExpression(
                                 field.ast.initializer,
                                 ctx,
-                                util,
+                                astF,
                             );
                         } else {
                             // if a field has optional type and it is missing an explicit initializer
                             // we consider it to be initialized with the null value
 
                             field.default =
                                 field.type.kind === "ref" && field.type.optional
-                                    ? util.makeNullLiteral(field.ast.loc)
+                                    ? getAstUtil(astF).makeNullLiteral(
```

Comment:
Maybe we can extract the `getAstUtil()` call out of the loops? To line 2216 for example